dev-docs/design-rationale.md

```diff
@@ -0,0 +1,49 @@
+
+## Side effects of design choices
+
+Software engineering is always about trade-offs and we had to make hard choices along the way.
+
+### Date and time handling
+
+In the v1 of the OSDK, we provided custom types for handling dates and times. To implement those types, we had to bundle a dependency on Luxon. Luxon is a great library for date/time math that uses immutable objects. However, there is also Momentjs for date/time math. Both of these libraries are massive in order to include all the possible timezone data across all time. This is problematic if you are already using one of those libraries and then are forced to include the other as you're paying for quite a lot of javascript to be loaded.
+
+There is also a javascript standard ([TC39 Temporal](https://github.com/tc39/proposal-temporal)) in the works that would remove the need for either of these libraries. That would mean that the user does not have to download large datetime libraries for every website they visit and the browser would keep the timezone information up to date.
+
+If the Temporal proposal was slightly further along, we would have shipped the SDK with a dependency on those type definitions and asked our consumers to polyfill for older browsers. This would mean that the SDK is quite future proof. However we are not quite there.
+
+Finally, there is a cost (albeit small) to parsing strings. It feels wrong to parse all of the date time data that is returned from the server into javascript objects if you may not even use it! 
+
+Therefore, we made the choice to not provide concrete types for date/time in the v2 of the OSDK and instead provide the raw strings in the international standard of [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601). This format is human readable allowing for simple inclusion in applications that do not need to process the data itself while being trivial to use with libraries like Luxon and Moment, ensuring there are not duplicative libraries that end-users need to pay the download/memory price for. 
+
+Additional reading:
+- [MomentJS project status (momentjs.com)](https://momentjs.com/docs/#/-project-status/) outlining why you shouldn't use moment.
+- [You Don't Need Momentjs](https://github.com/you-dont-need/You-Dont-Need-Momentjs/blob/master/README.md)
+
+### Discoverability 
+
+In v1, you might take a constructed client and say `client.ontology.objects.Car`, getting intellisense along the way. This style of discovery breaks tree shaking though so in v2, you get the discoverability from the import `import * as MyOSDK from "@my/osdk"`. Decent tree shaking tools will convert this to only importing what is needed while the IDE will present you with all the information.
```

Comment:
Should we maybe add that hitting `ctrl-space` will give you more intellisense suggestions as well? I wasn't actually aware of this myself until someone pointed it out to me, though I'm not sure if it would make sense to add that in the context of a design doc